src/VirtoCommerce.MarketingModule.Data/Handlers/CouponUsageRecordHandler.cs

```diff
@@ -78,6 +82,24 @@ private async Task RecordUsages(string objectId, IEnumerable<PromotionUsage> old
             }
         }
 
+        private async Task ValidateUsagesAndThrow(IEnumerable<PromotionUsage> usages)
```

Comment:
Using `IReadOnlyCollection<PromotionUsage>` is better, because prevent lazy evaluation for `usages`.

Reply:
ok